senders/client_factory.go

```diff
@@ -246,10 +246,10 @@ func Timeout(timeout time.Duration) Option {
 	}
 }
 
-func TLSConfigOptions(tlsCfg *tls.Config) Option {
+func TLSConfigOptions(tlsCfg tls.Config) Option {
```

Comment:
Please don't pass tlsCfg by value. I looked at the tls.Config type and it contains an unexported mutex. Mutexes cannot be safely copied by assignment.